package.xml

```diff
@@ -12,6 +12,7 @@
 
   <exec_depend condition="$ROS_PYTHON_VERSION == 2">python-mock</exec_depend>
   <exec_depend condition="$ROS_PYTHON_VERSION == 3">python3-mock</exec_depend>
+  <exec_depend condition="$ROS_PYTHON_VERSION == 3">python3-importlib-metadata</exec_depend>
```

Comment:
What happens for Python 2?

Reply:
Well, it actually looks like you can do a `pip install importlib_metadata` for Python 2 and have it actually work.  So we can either add a pip-only rosdep key for it, or remove support for Python 2 from this package.

Reply:
> we can either add a pip-only rosdep key for it, or remove support for Python 2 from this package.

A `pip` dependency would prohibit the package from being packaged into a Debian package.

I thought this was used somewhere in ROS 1 or tools but I can't seem to fine it. It doesn't seem to be released into any ROS 1 distro. Maybe @wjwwood can clarify.

Reply:
It's used by `catkin_tools`.

See http://packages.ros.org/ros/ubuntu/dists/bionic/main/binary-amd64/Packages:

```
Package: python-catkin-tools
Priority: optional
Section: python
Installed-Size: 786
Maintainer: William Woodall <william@osrfoundation.org>
Architecture: all
Version: 0.6.1-1
Depends: python-catkin-pkg (>= 0.2.9), python-osrf-pycommon, python-pkg-resources, python-trollius, python-yaml, python (<< 2.8), python (>= 2.7), python:any (>= 2.6.6-7~), python-argparse, python-setuptools
Conflicts: python3-catkin-tools

...

Package: python-osrf-pycommon
Priority: optional
Section: python
Installed-Size: 120
Maintainer: William Woodall <william@osrfoundation.org>
Architecture: all
Version: 0.1.9-1
Depends: python (<< 2.8), python (>= 2.7), python-pkg-resources, python-trollius, python:any (>= 2.6.6-7~), python-setuptools
```

Reply:
I don't know what this means for python2 support, but since we've essentially released the final python2 version of catkin tools we could drop python2 support here too, but it needs to be done properly.

I wouldn't 100% agree with merging this until that happens, though it might be ok, or this pull request is updated to preserve python2 support when that is appropriate.

Reply:
I did some work to remove Python 2 support completely for this package; you can see it in https://github.com/osrf/osrf_pycommon/tree/remove-python2 .

However, I suspect that is not exactly what you are talking about.  Could you explain a little more on how you see this going?  Should we go forward with the pip-only rosdep key and maintain Python 2 support here?  Should I remove Python 2 support from this package?  Should I *also* remove Python 2 support from catkin tools?  It's just not clear to me what the path forward is.  Thanks.

Reply:
`catkin_tools` has already had Python2 support removed. So we'd just need to remove the mechanisms in this package used to allow for support of both. I'll have a look at your branch.

Reply:
It looks ok, I'd have to go over it myself to know if it got most things, but for sure some of the process utils files should go away, some are Python2 specific others are Python3 specific.

Reply:
Sounds good, thanks for the feedback.  I'll open a PR with the changes in the remove-python2 branch, then we can iterate on that.  Once that is done and merged, I'll rebase this on top.